crates/block-producer/src/block_builder/prover/tests.rs

```diff
@@ -38,9 +36,9 @@ use crate::{
 /// The store will contain accounts 1 & 2, while the transaction batches will contain 2 & 3.
 #[test]
 fn block_witness_validation_inconsistent_account_ids() {
-    let account_id_1 = AccountId::new_unchecked(Felt::new(ACCOUNT_ID_OFF_CHAIN_SENDER));
-    let account_id_2 = AccountId::new_unchecked(Felt::new(ACCOUNT_ID_OFF_CHAIN_SENDER + 1));
-    let account_id_3 = AccountId::new_unchecked(Felt::new(ACCOUNT_ID_OFF_CHAIN_SENDER + 2));
+    let account_id_1 = AccountId::try_from(ACCOUNT_ID_OFF_CHAIN_SENDER).unwrap();
+    let account_id_2 = AccountId::try_from(ACCOUNT_ID_OFF_CHAIN_SENDER + (1 << 80)).unwrap();
+    let account_id_3 = AccountId::try_from(ACCOUNT_ID_OFF_CHAIN_SENDER + (2 << 80)).unwrap();
 
```

Comment:
The least significant bits of the new account ID must be zero, so we can't add low amounts. Plus, now for each account ID, the prefix (most significant 64 bits) must be modified. The 80 bits shift is so that the changed number ends up on the random bits and doesn't change the metadata.

Reply:
There must be a better way of generating arbitrary values for test purposes. Can we not add `rng` support for these types in `miden-base`?

This applies to many of the other types as well; where tests are awkward because just generating the fluff data is painful e.g. `TransactionId`.

Reply:
I double-checked and found a `new_dummy` method that is only available for testing so I changed the tests to use this instead.